src/api/resources.ts

```diff
@@ -44,3 +48,15 @@ export async function get(id: string) {
 
   return result
 }
+
+export async function createComment(
+  resourceId: number,
+  params: IResourceCommentParams,
+) {
+  const result = await apiClient
+    .post(`resources/${resourceId}/comments`, params)
```

Comment:
I think this might be your 422 problem. Generally rails expects a shape like this:
```
{ comment: { body: "abcd123" } }
```
Where the top-level key is the resource type, e.g. "comment". Try changing this to:
```suggestion
    .post(`resources/${resourceId}/comments`, { comment: params })
```